iqama_fetcher.py

```diff
@@ -1,42 +1,119 @@
-  post_params = {
-      'masjid': masjid,
-  }
-  print 'scraper get_params = ', post_params
-  request = requests.post(_SCRAPER_URL, params=post_params, timeout=15)
+  # making sure the masjid input is in UTF-8 format
+  masjid_encoded = util.EncodeParameter(masjid)
+  # getting the id from the _MASJID_METADATA
+  masjid_id = GetMasjidID(masjid_encoded)
+
+  desired_prayer = util.GetPrayerKeyName(desired_prayer)
+
+  print 'Iqamah.net Masjid ID ', masjid_id
+
+  # if masjid_id doesn't exist then return None
+  if not masjid_id:
+    return None
+
+  request = requests.get(_IQAMAH_NET_URL + str(masjid_id) + ".xml", timeout=15)
 
   # checks if request isn't bad
   # pylint: disable=no-member
   if request.status_code == requests.codes.ok:
-    response = json.loads(request.text)
-    print 'response from scraper service = ', response
-
-    # translate the dict keys to enums
-    iqama_times = {}
-    for key in response:
-      prayer = util.StringToDailyPrayer(key)
-      if prayer:
-        iqama_times[prayer] = response[key]
-
-    return iqama_times[desired_prayer]
-  print 'bad response from scraper service = ', request.status_code
+    # setting a tree node to the beginning of request xml content
+    tree = ElementTree.fromstring(request.content)
+    # setting an element node to element 'IqDay' which contains the
+    # prayer names and timings
+    # <IqDay><Fajr>##:##</Fajr>.....</IqDay>
+    iqamah_element = tree.findall('IqDay')[0]
+
+    # looping through the IqDay tags (prayer names)
+    for iqamah in iqamah_element:
+      # convert the prayer names in the IqDay element to a known format
+      iqamah_name = GetEquivalentPrayer(str(iqamah.tag))
+      if desired_prayer.lower() == (iqamah_name).lower():
+        iqama_time = iqamah.text
+        break
+
+    print 'Iqama Time ', iqama_time
+
+    # if iqama time was found then add AM or PM to the time
+    if iqama_time:
+      return AddAMPM(desired_prayer, str(iqama_time))
+    return None
+  print 'bad response from iqamah service = ', request.status_code
   return None
+
+def AddAMPM(prayer, time):
+  """Returns the time variable back with AM/PM added
+
+  Args:
+    prayer: a string representing desired salah
+    time: a string representing the iqamah time
+
+  Returns: a string containing the iqamah time along with AM/PM
+  """
+
+  # if the time input is in a format where it contains integers
+  # then proceed forward otherwise return None
+  if re.match(r'(\d+)', time, flags=0):
+    # strip the time variable from anything added to it
+    # example: 10:00 PM becomes 1000 and 5 AM becomes 5
+    time = ''.join(re.findall(r'(\d+)', time))
+    # if time length is more than 2 characters then add a : after
+    # the second character from the right
+    # example: 1000 becomes 10:00 and 5 stays 5
+    if len(time) > 2:
+      time = time[:1] + ':' + time[1:] 
+    # add PM by default except if the prayer is Fajr
+    if re.match(r'((^f)(.+)(r$))', prayer.lower(), flags=0):
```

Comment:
what if dhuhr is at 11 or something? there are time formatting that already do this, maybe we should use them? this seems hacky